### PR TITLE
Add packaging tasks library

### PR DESCRIPTION
**Pull request by joperezr**
cc: @weshaggard @ericstj @chcosta 

Adding the Packaging task library from buildtools which will be needed in order to onboard corefx and other repos into arcade. The way the csproj is set up doesn't require a nuspec file for the package to be generated.

I pushed initially 2 commits, so that you can diff commit 1 vs 2 to see the changes that went in to the targets files from the original copy from buildtools.

### CODE REVIEW

File: src/Microsoft.DotNet.Build.Tasks.Packaging/Microsoft.DotNet.Build.Tasks.Packaging.csproj [hunk 1]
```diff
@@ -0,0 +1,70 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFrameworks>netstandard2.0;net46</TargetFrameworks>
+    <CopyLocalLockFileAssemblies Condition="'$(TargetFramework)' == 'netstandard2.0'">true</CopyLocalLockFileAssemblies>
```
Comment (alexperovich) [inline src/Microsoft.DotNet.Build.Tasks.Packaging/Microsoft.DotNet.Build.Tasks.Packaging.csproj]: Why is this necessary? Why not just chain the "Publish" task before your "GatherTFMSpecificDllsOnPackage" task and use the publish output? See https://github.com/dotnet/arcade/blob/master/src/Microsoft.DotNet.Helix/Sdk/Microsoft.DotNet.Helix.Sdk.csproj#L51 for an example

File: src/Microsoft.DotNet.Build.Tasks.Packaging/Microsoft.DotNet.Build.Tasks.Packaging.csproj [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    <TargetsForTfmSpecificContentInPackage>$(TargetsForTfmSpecificContentInPackage);GatherTFMSpecificDllsOnPackage</TargetsForTfmSpecificContentInPackage>
+    <GenerateNuspecDependsOn>$(GenerateNuspecDependsOn);ClearBuildOutputInPackage</GenerateNuspecDependsOn>
```
Comment (alexperovich) [inline src/Microsoft.DotNet.Build.Tasks.Packaging/Microsoft.DotNet.Build.Tasks.Packaging.csproj]: Why not set `IncludeBuildOutput` to false rather than adding another target?

File: src/Microsoft.DotNet.Build.Tasks.Packaging/Microsoft.DotNet.Build.Tasks.Packaging.csproj [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    <TargetFilesLocation>$(MSBuildThisFileDirectory)build</TargetFilesLocation>
```
Comment (alexperovich) [inline src/Microsoft.DotNet.Build.Tasks.Packaging/Microsoft.DotNet.Build.Tasks.Packaging.csproj]: Should probably set `<PackageType>MSBuildSdk</PackageType>`

File: src/Microsoft.DotNet.Build.Tasks.Packaging/Microsoft.DotNet.Build.Tasks.Packaging.csproj [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  </PropertyGroup>
+
+  <!-- Dependencies that are specific to netstandard2.0 -->
+  <ItemGroup Condition="'$(TargetFramework)' == 'netstandard2.0'">
```
Comment (alexperovich) [inline src/Microsoft.DotNet.Build.Tasks.Packaging/Microsoft.DotNet.Build.Tasks.Packaging.csproj]: Can we use the same set of package references for both TFMs?

File: src/Microsoft.DotNet.Build.Tasks.Packaging/Microsoft.DotNet.Build.Tasks.Packaging.csproj [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    <PackageReference Include="Microsoft.NETCore.Platforms" Version="2.1.0" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <!-- Common Reference -->
+    <PackageReference Include="Newtonsoft.Json" Version="11.0.2" />
+    <PackageReference Include="NuGet.Commands" Version="4.7.0" />
+    <PackageReference Include="NuGet.Packaging" Version="4.7.0" />
+    <PackageReference Include="NuGet.ProjectModel" Version="4.7.0" />
+    <PackageReference Include="System.Reflection.Metadata" Version="1.6.0" />
```
Comment (alexperovich) [inline src/Microsoft.DotNet.Build.Tasks.Packaging/Microsoft.DotNet.Build.Tasks.Packaging.csproj]: You need to set PrivateAssets="All" on package references or they will be added as dependencies in the nuget package. See https://github.com/dotnet/arcade/blob/master/src/Microsoft.DotNet.Helix/Sdk/Microsoft.DotNet.Helix.Sdk.csproj#L35 for an example.

File: src/Microsoft.DotNet.Build.Tasks.Packaging/build/Microsoft.DotNet.Build.Tasks.Packaging.targets [hunk 6]
```diff
@@ -0,0 +1,7 @@
+<!-- Copyright (c) .NET Foundation. All rights reserved. -->
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="12.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+    <Import Project="Packaging.common.targets"/>
```
Comment (alexperovich) [inline src/Microsoft.DotNet.Build.Tasks.Packaging/build/Microsoft.DotNet.Build.Tasks.Packaging.targets]: NIT: 2 space indent

File: src/Microsoft.DotNet.Build.Tasks.Packaging/build/Microsoft.DotNet.Build.Tasks.Packaging.targets [hunk 7]
```diff
@@ -0,0 +1,7 @@
+<!-- Copyright (c) .NET Foundation. All rights reserved. -->
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="12.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
```
Comment (alexperovich) [inline src/Microsoft.DotNet.Build.Tasks.Packaging/build/Microsoft.DotNet.Build.Tasks.Packaging.targets]: Please remove the `ToolsVersion` and xmlns from the targets files. They aren't necessary anymore.

File: src/Microsoft.DotNet.Build.Tasks.Packaging/Microsoft.DotNet.Build.Tasks.Packaging.csproj (hunk 2 above)
[the remark this answers is not in the document]
Comment (joperezr) [reply src/Microsoft.DotNet.Build.Tasks.Packaging/Microsoft.DotNet.Build.Tasks.Packaging.csproj]: This was needed becase I did want to run before GenerateNuspec, but after the other one, which is why I did it like this. That said, I'm about to push some changes that will basically use the arcade's way of packaging, using a nuspec instead so that we are consistent with the rest of the repo.

File: src/Microsoft.DotNet.Build.Tasks.Packaging/Microsoft.DotNet.Build.Tasks.Packaging.csproj (hunk 1 above)
Comment (joperezr) [reply src/Microsoft.DotNet.Build.Tasks.Packaging/Microsoft.DotNet.Build.Tasks.Packaging.csproj]: Removed since I have now moved to using nuspec instead.

File: src/Microsoft.DotNet.Build.Tasks.Packaging/Microsoft.DotNet.Build.Tasks.Packaging.csproj (hunk 2 above)
Comment (joperezr) [reply src/Microsoft.DotNet.Build.Tasks.Packaging/Microsoft.DotNet.Build.Tasks.Packaging.csproj]: Removed.

File: src/Microsoft.DotNet.Build.Tasks.Packaging/Microsoft.DotNet.Build.Tasks.Packaging.csproj (hunk 3 above)
Comment (joperezr) [reply src/Microsoft.DotNet.Build.Tasks.Packaging/Microsoft.DotNet.Build.Tasks.Packaging.csproj]: I haven't set this one, but I'm not yet 100% sure if this will become an SDK package.

File: src/Microsoft.DotNet.Build.Tasks.Packaging/Microsoft.DotNet.Build.Tasks.Packaging.csproj (hunk 5 above)
Comment (joperezr) [reply src/Microsoft.DotNet.Build.Tasks.Packaging/Microsoft.DotNet.Build.Tasks.Packaging.csproj]: I didn't have to do this since I'm now using a nuspec.

File: src/Microsoft.DotNet.Build.Tasks.Packaging/build/Packaging.common.targets [hunk 8]
```diff
@@ -0,0 +1,52 @@
+<!-- Copyright (c) .NET Foundation. All rights reserved. -->
+<?xml version="1.0" encoding="utf-8"?>
+<Project>
+  <PropertyGroup>
+    <_PackagingCommonTargetsImported>true</_PackagingCommonTargetsImported>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <_TaskLibraryDir Condition="'$(_TaskLibraryDir)' == '' AND '$(RunningOnCore)' == 'true'">$(MSBuildThisFileDirectory)netstandard2.0/</_TaskLibraryDir>
```
Comment (ericstj) [inline src/Microsoft.DotNet.Build.Tasks.Packaging/build/Packaging.common.targets]: Please revert this property rename.  It should be a public property as it was before.

Comment (joperezr) [reply src/Microsoft.DotNet.Build.Tasks.Packaging/build/Packaging.common.targets]: Will do, I didn't go and check in corefx to see if somebody else depended on the property name, which is why I decided to introduce a new one, but I agree that it is best to revert and use the original name.

File: src/Microsoft.DotNet.Build.Tasks.Packaging/Microsoft.DotNet.Build.Tasks.Packaging.csproj [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+  <PropertyGroup>
+    <TargetFrameworks>netcoreapp2.0;net46</TargetFrameworks>
+    <CopyLocalLockFileAssemblies Condition="'$(TargetFramework)' == 'netcoreapp2.0'">true</CopyLocalLockFileAssemblies>
+    <IsPackable>true</IsPackable>
+    <NuspecFile>$(MSBuildProjectName).nuspec</NuspecFile>
+    <NuspecBasePath>$(MSBuildProjectDirectory)</NuspecBasePath>
+  </PropertyGroup>
+
+  <!-- Dependencies that are specific to netcoreapp2.0 -->
+  <ItemGroup Condition="'$(TargetFramework)' == 'netcoreapp2.0'">
```
Comment (tmat) [inline src/Microsoft.DotNet.Build.Tasks.Packaging/Microsoft.DotNet.Build.Tasks.Packaging.csproj]: ><ItemGroup Condition="'$(TargetFramework)' == 'netcoreapp2.0'"> [](start = 2, length = 63)

Conditional dependencies should not be necessary. 
I'd target netcoreapp2.0 and net461 btw.

File: src/Microsoft.DotNet.Build.Tasks.Packaging/Microsoft.DotNet.Build.Tasks.Packaging.nuspec [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+    <file src="$ArtifactsBinDir$Microsoft.DotNet.Build.Tasks.Packaging\net46\NuGet.Versioning.dll" target="build\net46\NuGet.Versioning.dll" />
+    <file src="$ArtifactsBinDir$Microsoft.DotNet.Build.Tasks.Packaging\net46\System.Collections.Immutable.dll" target="build\net46\System.Collections.Immutable.dll" />
+    <file src="$ArtifactsBinDir$Microsoft.DotNet.Build.Tasks.Packaging\net46\System.Reflection.Metadata.dll" target="build\net46\System.Reflection.Metadata.dll" />
+    <file src="F:\scratch\packagingtemp\bin\Debug\netcoreapp2.0\Microsoft.Build.Framework.dll" target="build\netcoreapp2.0\Microsoft.Build.Framework.dll" />
+    <file src="F:\scratch\packagingtemp\bin\Debug\netcoreapp2.0\Microsoft.Build.Utilities.Core.dll" target="build\netcoreapp2.0\Microsoft.Build.Utilities.Core.dll" />
+    <file src="F:\scratch\packagingtemp\bin\Debug\netcoreapp2.0\Microsoft.DotNet.Build.Tasks.Packaging.dll" target="build\netcoreapp2.0\Microsoft.DotNet.Build.Tasks.Packaging.dll" />
+    <file src="F:\scratch\packagingtemp\bin\Debug\netcoreapp2.0\Newtonsoft.Json.dll" target="build\netcoreapp2.0\Newtonsoft.Json.dll" />
```
Comment (tmat) [inline src/Microsoft.DotNet.Build.Tasks.Packaging/Microsoft.DotNet.Build.Tasks.Packaging.nuspec]: >F:\scratch\packagingtemp\bin [](start = 15, length = 28)

Huh?

Comment (joperezr) [reply src/Microsoft.DotNet.Build.Tasks.Packaging/Microsoft.DotNet.Build.Tasks.Packaging.nuspec]: duh, I see what's wrong now, it took me a while!

File: src/Microsoft.DotNet.Build.Tasks.Packaging/build/Packaging.targets [hunk 11]
```diff
@@ -0,0 +1,1309 @@
+<!-- Copyright (c) .NET Foundation. All rights reserved. -->
+<?xml version="1.0" encoding="utf-8"?>
+<Project>
+  <Import Condition="'$(_PackagingCommonTargetsImported)' != 'true'" Project="Packaging.common.targets"/>
```
Comment (tmat) [inline src/Microsoft.DotNet.Build.Tasks.Packaging/build/Packaging.targets]: >c [](start = 88, length = 1)

Common (capital 'C')

File: src/Microsoft.DotNet.Build.Tasks.Packaging/build/Packaging.targets [hunk 12]
```diff
@@ -0,0 +1,1309 @@
...
+       to false for the Beta/RC builds that get uploaded to NuGet.
+  -->
+  <PropertyGroup>
+    <PackageVersion Condition="'$(StableVersion)' != ''">$(StableVersion)</PackageVersion>
+    <PackageVersion Condition="'$(PackageVersion)' == '' AND '$(Version)' != ''">$(Version)</PackageVersion>
+    <PreReleaseLabel Condition="'$(PreReleaseLabel)' == ''">rc2</PreReleaseLabel>
+    <IncludeBuildNumberInPackageVersion Condition="'$(IncludeBuildNumberInPackageVersion)' == ''">true</IncludeBuildNumberInPackageVersion>
+
+    <VersionSuffix Condition="'$(PreReleaseLabel)' != ''">-$(PreReleaseLabel)</VersionSuffix>
+    <VersionSuffix Condition="'$(IncludeBuildNumberInPackageVersion)' == 'true'">$(VersionSuffix)-$(BuildNumberMajor)-$(BuildNumberMinor)</VersionSuffix>
```
Comment (tmat) [inline src/Microsoft.DotNet.Build.Tasks.Packaging/build/Packaging.targets]: Why do we need this? Versioning is handled in RepoToolset.

Comment (joperezr) [reply src/Microsoft.DotNet.Build.Tasks.Packaging/build/Packaging.targets]: I believe they only handle assembly versioning but corefx has its own logic for versioning on the packages.

Comment (weshaggard) [reply src/Microsoft.DotNet.Build.Tasks.Packaging/build/Packaging.targets]: We will be switching to using the common Versioning coming from Arcade but we haven't done that yet. This is just an incremental refactoring to get it into a consumable state in corefx and then we will adjust once we start consuming the versioning from here as well.

File: src/Microsoft.DotNet.Build.Tasks.Packaging/Microsoft.DotNet.Build.Tasks.Packaging.nuspec [hunk 13]
```diff
@@ -0,0 +1,95 @@
...
+- VerifyTypes
+    </description>
+    <copyright>Copyright © Microsoft Corporation</copyright>
+    <tags></tags>
+    <dependencies>
+       <!-- Do not add any dependencies to this package as we want to install it versionless (i.e. -ExcludeVersion) and we don't want the dependencies to be installed that way -->
+    </dependencies>
+  </metadata>
+  <files>
+    <file src="$ArtifactsBinDir$Microsoft.DotNet.Build.Tasks.Packaging\net46\Microsoft.DotNet.Build.Tasks.Packaging.dll" target="build\net46\Microsoft.DotNet.Build.Tasks.Packaging.dll" />
```
Comment (weshaggard) [inline src/Microsoft.DotNet.Build.Tasks.Packaging/Microsoft.DotNet.Build.Tasks.Packaging.nuspec]: Do we really need to manually maintain this file list? Perhaps I misunderstood you but I thought you said you got dotnet pack to work without needing to manually create the nuspec.

File: src/Microsoft.DotNet.Build.Tasks.Packaging/build/Microsoft.DotNet.Build.Tasks.Packaging.targets [hunk 14]
```diff
@@ -0,0 +1,7 @@
+<!-- Copyright (c) .NET Foundation. All rights reserved. -->
+<?xml version="1.0" encoding="utf-8"?>
+<Project>
+  <Import Project="Packaging.common.targets"/>
```
Comment (weshaggard) [inline src/Microsoft.DotNet.Build.Tasks.Packaging/build/Microsoft.DotNet.Build.Tasks.Packaging.targets]: Is there any good reason these are split into multiple targets file given they will always be imported? Perhaps just for separation?

File: src/Microsoft.DotNet.Build.Tasks.Packaging/README.md [hunk 15]
```diff
@@ -0,0 +1,3 @@
+# Microsoft.DotNet.Build.Tasks.Packaging
+
+Task package which will handle building projects with the extension pkgproj like the ones we currently have on our [corefx](http://github.com/dotnet/corefx/) and [core-setup](http://github.com/dotnet/core-setup/) repos. This package also includes the tasks that will do validation to make sure the packages will not break support of any existing framework.
```
Comment (weshaggard) [inline src/Microsoft.DotNet.Build.Tasks.Packaging/README.md]: This package is still very corefx specific and I"m not sure anyone else will be able to use it anytime soon without a lot more documentation and/or simplification.

File: src/Microsoft.DotNet.Build.Tasks.Packaging/build/Microsoft.DotNet.Build.Tasks.Packaging.targets (hunk 14 above)
Comment (joperezr) [reply src/Microsoft.DotNet.Build.Tasks.Packaging/build/Microsoft.DotNet.Build.Tasks.Packaging.targets]: That is probably a better question for @ericstj in terms of why they where like this on the beginning, but at least I kept them like that for separation, It was nice to have all targets specific to the libs on one file and all the general other ones for packaging, nuspec generation, validation, etc on the other.

File: src/Microsoft.DotNet.Build.Tasks.Packaging/README.md (hunk 15 above)
Comment (joperezr) [reply src/Microsoft.DotNet.Build.Tasks.Packaging/README.md]: Agreed, well, not just corefx but wcf and core-setup.

Comment (joperezr) [reply src/Microsoft.DotNet.Build.Tasks.Packaging/README.md]: Do you want me to add a comment for that on the readme or the package description?

File: src/Microsoft.DotNet.Build.Tasks.Packaging/build/Microsoft.DotNet.Build.Tasks.Packaging.targets (hunk 14 above)
Comment (ericstj) [reply src/Microsoft.DotNet.Build.Tasks.Packaging/build/Microsoft.DotNet.Build.Tasks.Packaging.targets]: Yep, one is for pkgproj projects, another is for content (csproj, deproj, etc), and one is shared.

File: src/Microsoft.DotNet.Build.Tasks.Packaging/Microsoft.DotNet.Build.Tasks.Packaging.csproj [hunk 16]
```diff
@@ -37,6 +73,51 @@
...
+    <Content Include="$(MSBuildThisFileDirectory)..\..\THIRD-PARTY-NOTICES.TXT">
+      <PackagePath>THIRD-PARTY-NOTICES.TXT</PackagePath>
+      <Pack>true</Pack>
+    </Content>
+  </ItemGroup>
+
+  <!-- When RAR runs it removes the metadata we set to mscorlib reference so that it does not get includded
+  as a Framework Assembly Reference in the package. This seems to only happen with mscorlib. For that
+  reason, as a workaround we readd the metadata back with this target after RAR runs. -->
+  <Target Name="RemoveMscorlibFromPackageReferences" AfterTargets="ResolveAssemblyReferences">
```
Comment (ericstj) [inline src/Microsoft.DotNet.Build.Tasks.Packaging/Microsoft.DotNet.Build.Tasks.Packaging.csproj]: I think you can set NoStdLib to control this: https://github.com/Microsoft/msbuild/blob/6b3237d46cd46bafdd8db69c9811be8cee4ceb10/src/Tasks/Microsoft.CSharp.CurrentVersion.targets#L364

Comment (joperezr) [reply src/Microsoft.DotNet.Build.Tasks.Packaging/Microsoft.DotNet.Build.Tasks.Packaging.csproj]: Setting the property doesn't help. The reference to mscorlib is not coming from the platform itself, but instead is coming from the targeting pack package that gets automatically added by the SDK project when you target net46.

**Comment by joperezr** [comment]
@weshaggard I moved it back to not use a nuspec and use dotnet pack again. Is this good to merge?

File: src/Microsoft.DotNet.Build.Tasks.Packaging/Microsoft.DotNet.Build.Tasks.Packaging.csproj (hunk 5 above)
Comment (ViktorHofer) [reply src/Microsoft.DotNet.Build.Tasks.Packaging/Microsoft.DotNet.Build.Tasks.Packaging.csproj]: I read your comment that you removed the nuspec again, do you now need the PrivateAssets switch?

Comment (joperezr) [reply src/Microsoft.DotNet.Build.Tasks.Packaging/Microsoft.DotNet.Build.Tasks.Packaging.csproj]: Yup, I added the switch as well

File: src/Microsoft.DotNet.Build.Tasks.Packaging/Microsoft.DotNet.Build.Tasks.Packaging.csproj (hunk 16 above)
Comment (ericstj) [reply src/Microsoft.DotNet.Build.Tasks.Packaging/Microsoft.DotNet.Build.Tasks.Packaging.csproj]: `targeting pack package`- what's this?  Specifically what package.  Its busted if the SDK is auto-referencing anything that isn't disablable.

Comment (joperezr) [reply src/Microsoft.DotNet.Build.Tasks.Packaging/Microsoft.DotNet.Build.Tasks.Packaging.csproj]: Package is: `Microsoft.NETFramework.ReferenceAssemblies/1.0.0-alpha-003`

Comment (ericstj) [reply src/Microsoft.DotNet.Build.Tasks.Packaging/Microsoft.DotNet.Build.Tasks.Packaging.csproj]: I still assert that its busted that the package is adding mscorlib when it shouldn't.  It needs to honor the same properties that the regular targets/targeting packs do.

Here's my best guess at where it came from, but you'll have a better idea if you look at the /pp output. /cc @dsplaisted 
https://github.com/dsplaisted/ReferenceAssemblyPackages/blob/173de49c73219af5750e6ac238a707a0e991e541/Microsoft.NETFramework.ReferenceAssemblies/Microsoft.NETFramework.ReferenceAssemblies.targets#L61

Comment (dsplaisted) [reply src/Microsoft.DotNet.Build.Tasks.Packaging/Microsoft.DotNet.Build.Tasks.Packaging.csproj]: @ericstj, Yes, the reference assembly package handles mscorlib a bit differently since the built-in targets don't entirely handle it correctly when the TargetFrameworkRootPath is overridden.  The targets from that package can be updated to respect `NoStdLib`.  I've [updated](https://github.com/dsplaisted/ReferenceAssemblyPackages/commit/f6146b8e9d5e7800ef1986c96cc5dd818320386b) the implicit reference to set Pack to false, but that might not fix the problem until https://github.com/dotnet/sdk/issues/2254 is fixed.

BTW, I'd recommend using a .binlog over a /pp output first for finding where something comes from in a build.

**Comment by joperezr** [comment]
@dotnet-bot test this please

File: src/Microsoft.DotNet.Build.Tasks.Packaging/src/AssemblyResolver.cs [hunk 17]
```diff
@@ -0,0 +1,109 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Reflection;
+
+namespace Microsoft.DotNet.Build.Common.Desktop
```
Comment (weshaggard) [inline src/Microsoft.DotNet.Build.Tasks.Packaging/src/AssemblyResolver.cs]: We likely want to eventually move this to some common location so we can address this in all the tasks see https://github.com/dotnet/arcade/issues/47.

File: src/Microsoft.DotNet.Build.Tasks.Packaging/Microsoft.DotNet.Build.Tasks.Packaging.csproj [hunk 18]
```diff
@@ -0,0 +1,178 @@
...
+    <ItemGroup>
+      <ReferencePath Update="@(ReferencePath)">
+        <Pack>false</Pack>
+      </ReferencePath>
+    </ItemGroup>
+  </Target>
+
```
Comment (weshaggard) [inline src/Microsoft.DotNet.Build.Tasks.Packaging/Microsoft.DotNet.Build.Tasks.Packaging.csproj]: I'm not going to claim to fully understand all the hacking going on here to make pack work but can we please make sure that we have the correct issues filed to get them fixed in the shipping SDK? Also if this is a pattern plan to use for all the packages in arcade we should refactor this into a targets file that we share so it doesn't get duplicated.

File: src/Microsoft.DotNet.Build.Tasks.Packaging/build/Microsoft.DotNet.Build.Tasks.Packaging.targets (hunk 14 above)
Comment (weshaggard) [reply src/Microsoft.DotNet.Build.Tasks.Packaging/build/Microsoft.DotNet.Build.Tasks.Packaging.targets]: If that is the case what will be our strategy to import the appropriate targets file in each of the context now?

File: src/Microsoft.DotNet.Build.Tasks.Packaging/Microsoft.DotNet.Build.Tasks.Packaging.csproj (hunk 18 above)
Comment (alexperovich) [reply src/Microsoft.DotNet.Build.Tasks.Packaging/Microsoft.DotNet.Build.Tasks.Packaging.csproj]: This really should be done as part of #69. That issue is part of post arcade v1 work which is odd to me though.

File: src/Microsoft.DotNet.Build.Tasks.Packaging/Microsoft.DotNet.Build.Tasks.Packaging.csproj (hunk 16 above)
Comment (ericstj) [reply src/Microsoft.DotNet.Build.Tasks.Packaging/Microsoft.DotNet.Build.Tasks.Packaging.csproj]: >  I'd recommend using a .binlog over a /pp output first

Each have their own place.  I was just trying to make it clear that I hadn't looked at more than the PR so @joperezr should check my work.

File: src/Microsoft.DotNet.Build.Tasks.Packaging/build/Microsoft.DotNet.Build.Tasks.Packaging.targets (hunk 14 above)
Comment (joperezr) [reply src/Microsoft.DotNet.Build.Tasks.Packaging/build/Microsoft.DotNet.Build.Tasks.Packaging.targets]: Packagelibs.targets and packaging.common.targets are imported everywhere, the Packaging.targets are only imported when project is a pkgproj.

File: src/Microsoft.DotNet.Build.Tasks.Packaging/src/AssemblyResolver.cs (hunk 17 above)
Comment (joperezr) [reply src/Microsoft.DotNet.Build.Tasks.Packaging/src/AssemblyResolver.cs]: Sounds good, I'll take care of this on my next PR in which I'll bring over test stuff from buildtools over.

File: src/Microsoft.DotNet.Build.Tasks.Packaging/Microsoft.DotNet.Build.Tasks.Packaging.csproj (hunk 18 above)
Comment (joperezr) [reply src/Microsoft.DotNet.Build.Tasks.Packaging/Microsoft.DotNet.Build.Tasks.Packaging.csproj]: I agree, I'll add a comment there saying that this snippet has to be part of that sdk.

@weshaggard the only thing we are doing in this ItemGroup is to make sure that the resulting package doesn't have any package or platform references as we are packing all of the required references inside the package itself.

**Comment by joperezr** [comment]
I'll go ahead and merge this in. I'll address any additional feedback on a subsequent PR.
